apps/system/js/app_install_manager.js

```diff
@@ -256,6 +261,57 @@
         this.authorUrl.textContent = '';
       }
 
+      // Warn about system level addons?
+      var warningEl = document.getElementById('system-addon-warning');
+      var showWarning = false;
+
+      if (manifest.content_scripts) {
+        var contentScripts = manifest.content_scripts;
+        var patterns = contentScripts.map(contentScript => {
+          return {
+            matches: new MatchPattern(contentScript.matches),
+            excludeMatches: new MatchPattern(contentScript.exclude_matches)
+          };
+        });
+
+        appLoop:
```

Comment:
let's move the app matching logic out to a separate function, because i think it's much more readable to have something like:
```
var showWarning = this.matchesHomescreenOrSystem(patterns);
```
or maybe even:
```
var warningEl = document.getElementById('system-addon-warning');
warningEl.hidden = !this.requiresRestart(manifest.content_scripts);